src/category_theory/abelian/abelian.lean

```diff
@@ -0,0 +1,554 @@
+protected abbreviation coimage : C := cokernel (kernel.ι f)
+
+/-- The projection onto the coimage. -/
+protected abbreviation coimage.π : P ⟶ abelian.coimage f :=
+cokernel.π (kernel.ι f)
+
+/-- There is a canonical monomorphism `i : coimage f ⟶ Q`. -/
+protected abbreviation factor_thru_coimage : abelian.coimage f ⟶ Q :=
+cokernel.desc (kernel.ι f) f $ kernel.condition f
+
+/-- `f` factors through its coimage via the canonical morphism `p`. -/
+protected lemma coimage.fac : coimage.π f ≫ abelian.factor_thru_coimage f = f :=
+cokernel.π_desc _ _ _
+
+/-- The canonical morphism `i : coimage f ⟶ Q` is a monomorphism -/
+instance : mono (abelian.factor_thru_coimage f) :=
+let I := abelian.coimage f, i := abelian.factor_thru_coimage f, p := cokernel.π (kernel.ι f) in
+mono_of_cancel_zero _ $ λ R (g : R ⟶ I) (hgi : g ≫ i = 0),
+begin
+  -- Since C is abelian, u := p ≫ coker g is the cokernel of some morphism h.
+  let u := p ≫ cokernel.π g,
+  haveI : epi u := epi_comp _ _,
+  have hu := abelian.normal_epi u,
+  let h := hu.g,
+
+  -- By hypothesis, i factors through the cokernel of g via some t.
+  obtain ⟨t, ht⟩ := cokernel.desc' g i hgi,
+
+  have hf : h ≫ f = 0, calc
+    h ≫ f = h ≫ (p ≫ i) : (coimage.fac f).symm ▸ rfl
+    ... = h ≫ (p ≫ (cokernel.π g ≫ t)) : ht ▸ rfl
+    ... = h ≫ u ≫ t : by simp only [category.assoc]; conv_lhs { congr, skip, rw ←category.assoc }
+    ... = 0 ≫ t : by rw [←category.assoc, hu.w]
+    ... = 0 : has_zero_morphisms.zero_comp _ _,
+
+  -- h factors through the kernel of f via some l.
+  obtain ⟨l, hl⟩ := kernel.lift' f h hf,
+
+  have hhp : h ≫ p = 0, calc
+    h ≫ p = (l ≫ kernel.ι f) ≫ p : hl ▸ rfl
+    ... = l ≫ 0 : by rw [category.assoc, cokernel.condition]
+    ... = 0 : has_zero_morphisms.comp_zero _ _,
+
+  -- p factors through u = coker h via some s.
+  obtain ⟨s, hs⟩ := normal_epi.desc' u p hhp,
+
+  have hs' : p ≫ cokernel.π g ≫ s = p ≫ 𝟙 I, by rw [←category.assoc, hs, category.comp_id],
+
+  haveI : mono (cokernel.π g) := mono_of_mono_fac ((cancel_epi _).1 hs'),
+
+  -- coker g is a monomorphism, but g ≫ coker g = 0 = 0 ≫ coker g, so g = 0 as required.
+  exact zero_of_comp_mono _ (cokernel.condition g)
+end
+
+instance epi_factor_thru_coimage [epi f] : epi (abelian.factor_thru_coimage f) :=
+epi_of_epi_fac $ coimage.fac f
+
+instance is_iso_factor_thru_coimage [epi f] : is_iso (abelian.factor_thru_coimage f) :=
+is_iso_of_mono_of_epi _
+
+/-- Factoring through the coimage is a strong epi-mono factorisation. -/
+@[simps] def coimage_strong_epi_mono_factorisation : strong_epi_mono_factorisation f :=
+{ I := abelian.coimage f,
+  m := abelian.factor_thru_coimage f,
+  m_mono := by apply_instance,
+  e := coimage.π f,
+  e_strong_epi := strong_epi_of_epi _ }
+
+end factor
+
+section has_strong_epi_mono_factorisations
+
+@[priority 100] instance : has_strong_epi_mono_factorisations.{v} C :=
+⟨λ X Y f, image_strong_epi_mono_factorisation f⟩
+
```

Comment:
I think it would be useful to add a reminder here about the consequences of this instance, in particular that this provides `has_images C` and `has_image_maps C`.